.github/workflows/continuous-integration.yml

```diff
@@ -47,12 +47,13 @@ jobs:
       - name: Install non-Python dependencies on macOS
         if: matrix.os == 'macos-latest'
         run: |
+          brew reinstall gcc
           brew install open-mpi
           brew install hdf5-mpi
           brew install libomp
-          if [[ ! -f "/usr/local/bin/gfortran" ]]; then
-            gfort=$(ls /usr/local/bin/gfortran-* | tail -n 1)
-            ln -s ${gfort} /usr/local/bin/gfortran
+          if ! command -v gfortran; then
+            echo "gfortran could not be found"
+            exit 1
```

Comment:
The macOS runners provided by GitHub already provide multiple version of the GFortran compiler, but none of them is actually called `gfortran` (see [here](https://github.com/actions/runner-images/blob/7781af869bec02d4d491cba6d01de412e96d865d/images/macos/macos-14-Readme.md?plain=1#L16-L18)).

Instead of reinstalling GCC, I believe that we can replicate [the strategy adopted in Pyccel](https://github.com/pyccel/pyccel/blob/1e97f1b6a21d226fc5863853252e8f720178bac6/.github/actions/macos_install/action.yml#L10-L18), where we find the newest version of GFortran and create a symbolic link to it called `gfortran`:
```yaml
        GFORTRAN_HOME=$(which gfortran || true)
        echo "GFORTRAN_HOME : $GFORTRAN_HOME"
        if [[ ! -f "$GFORTRAN_HOME" ]]; then
          gfort=$(find ${PATH//:/\/ } -name 'gfortran-*' -exec basename {} \; | sort | tail -n 1 || true)
          echo "Found $gfort"
          gfort_path=$(which ${gfort})
          folder=$(dirname ${gfort_path})
          ln -s ${gfort_path} ${folder}/gfortran
        fi
```

Reply:
Implemented in [1f8f8cb](https://github.com/pyccel/psydac/pull/407/commits/1f8f8cb4d828c87fc9295701c8b9974430821f79)